components/feature/readerview/src/main/java/mozilla/components/feature/readerview/ReaderViewFeature.kt

```diff
@@ -65,46 +61,19 @@ class ReaderViewFeature(
     internal var extensionController = WebExtensionController(READER_VIEW_EXTENSION_ID, READER_VIEW_EXTENSION_URL)
 
     @VisibleForTesting
-    internal val config = Config(context.getSharedPreferences(SHARED_PREF_NAME, Context.MODE_PRIVATE))
+    internal val config = ReaderViewConfig(
+        prefs = context.getSharedPreferences(SHARED_PREF_NAME, Context.MODE_PRIVATE)
```

Comment:
I assume the call to `getSharedPreferences` is fine and doesn't cause a violation? It's only the `getString`/`getInt` calls?

Reply:
I'm not sure, cc @mcomella

Reply:
It should give you  `onDiskRead` violation. There is a `StrictMode.allowThreadPolicy.resetAfter` function you can use if moving this to a IO coroutine is not possible.

Reply:
@NotWoods Your goal here is to *not* read from shared prefs when the fragment (and feature) is created, right? Instead wait until its needed and the appearance controls are used. I think that's good enough, because appearance controls won't be perf. sensitive (and are rarely used, if ever).

So, probably not worth the complexity of moving to an IO thread in this case.

Reply:
Yeah, my intent here is just to move the logic away from startup. I refactored the code to make `getSharedPreferences` lazy as well.

Reply:
I was about to suggest making this a `lazy` ref., but you already did that in your latest change. :).

Let me do a round of testing, but otherwise, I think this can land.